iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/pointer_repository.hpp

```diff
@@ -101,7 +101,6 @@ class PointerRepository
     id_t searchId(ptr_t ptr) const noexcept;
 
   private:
-    /// @todo: if required protect vector against concurrent modification
```

Comment:
The following lines belong to the todo as well. I'm not sure if we should remove the todo. Maybe we could add it as a low prio task to #605  :thinking:

Reply:
@FerdinandSpitzschnueffler @elfenpiff I would not add this to #605 but replace it with a comment, that the `PointerRepository` is not thread-safe.

Reply:
Not threadsafe? This might be a problem. We have to check if memory synchronization issues can arise.

Scenario:
- thread A is started
- thread B is started
- thread A registers a pointer
- thread B registers a pointer -> is this a valid use case
- thread B uses a relative pointer -> do we need to sync the memory

cc @MatthiasKillat maybe a whiteboard session at some time

Reply:
@FerdinandSpitzschnueffler @mossmaurice we only mark thread-safe classes with a comment that they are thread-safe. All other classes are by default non-thread-safe.
If we start this here we may start a fire ...

Reply:
@elBoberido this is your first and last warning. You are in vacation!

Maybe this was just a trap to lure you out of your cosy hammock. The next thing you should do is go to the next bar with your laptop and trade it against at least 5 cocktails or maybe 6. :cocktail:

Reply:
@elfenpiff that's not quite correct. My vacation starts tomorrow ;)

Reply:
@mossmaurice @FerdinandSpitzschnueffler @MatthiasKillat  ( and NOT @elBoberido  )

I created an issue: https://github.com/eclipse-iceoryx/iceoryx/issues/1701 to track this but I would define it out of scope for this PR.

Reply:
Fine for me but I still think that the following lines belong to the removed comment. Either remove them as well or replace the `todo` with `note`.